src/main/java/com/hcl/appscan/jenkins/plugin/scanners/Scanner.java

```diff
@@ -45,18 +53,28 @@ public boolean getHasOptions() {
 	public String getTarget() {
 		return m_target;
 	}
-        
-	public boolean isRescan() {
+
+	@DataBoundSetter
+	public void setRescan(boolean rescan) {
+		m_rescan = rescan;
+	}
+
+	public boolean getRescan() {
 		return m_rescan;
 	}
-        
+
+	@DataBoundSetter
+	public void setScanId(String scanId) {
+		m_scanId = scanId;
+	}
+
 	public String getScanId() {
 		return m_scanId;
 	}
 	
 	public abstract Map<String, String> getProperties(VariableResolver<String> resolver) throws AbortException;
 
-	public abstract void validateSettings(JenkinsAuthenticationProvider authProvider, Map<String, String> properties, IProgress progress) throws AbortException;
+	public abstract void validateScannerSettings(JenkinsAuthenticationProvider authProvider, Map<String, String> properties, IProgress progress, boolean isAppScan360) throws AbortException;
```

Comment:
Rather than creating this abstract method that subclasses are forced to implement, you can keep the existing validateSettings() method. The scanners that need to add their own validation can override the method, call super.validateSettings() and then do their own validation logic.

Reply:
Okay, sure.

Reply:
Can this update be made?

Reply:
This comment has already been addressed in the commit: https://github.com/jenkinsci/appscan-plugin/pull/236/commits/0ae604f59b1dd53a11fee8e684ec583b0a0998e0.
Now, I have removed the common abstract method from the base class. All the scan type classes overrides the existing validateSettings() method.